File: src/main/java/org/shark/example/service/mapstruct/GetMapStructExampleResponseService.java
package org.shark.example.service.mapstruct;

import lombok.RequiredArgsConstructor;
import org.shark.example.controller.mapstruct.pojo.MapStructResponseDto;
import org.shark.example.controller.mapstruct.pojo.MapStructResponseMapper;
import org.shark.example.service.mapstruct.pojo.MapStructExampleServiceDto;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor
@Service
public class GetMapStructExampleResponseService {

    private final MapStructResponseMapper mapStructResponseMapper;

    public MapStructResponseDto start() {
        MapStructExampleServiceDto serviceDto = new MapStructExampleServiceDto();
        serviceDto.setServiceValue1("value1");
        serviceDto.setServiceValue2("value2");
        serviceDto.setServiceValue3("value3");
        return mapStructResponseMapper.toResponse(serviceDto);
    }
}

File: src/main/kotlin/org/shark/example/service/mapstruct/GetMapStructExampleResponseService.kt
package org.shark.example.service.mapstruct

import org.shark.example.controller.mapstruct.pojo.MapStructResponseDto
import org.shark.example.controller.mapstruct.pojo.MapStructResponseMapper
import org.shark.example.service.mapstruct.pojo.MapStructExampleServiceDto
import org.springframework.stereotype.Service

@Service
class GetMapStructExampleResponseService(
    private val mapStructResponseMapper: MapStructResponseMapper
) {

    fun start(): MapStructResponseDto {
        val serviceDto = MapStructExampleServiceDto()
        serviceDto.serviceValue1 = "value1"
        serviceDto.serviceValue2 = "value2"
        serviceDto.serviceValue3 = "value3"
        return mapStructResponseMapper.toResponse(serviceDto)
    }
}
